alphagov/paas-product-pages
# Migrate from Jest to Vitest

## What 

Replace Jest with [Vitest](https://vitest.dev/) including new config files, settings, test update and dependency updates

## Why

Vite is quicker, works with ES modules and I'm using this as testing ground to do the work on the much more complex paas-admin migration

# How to test

- the test pass
- [run locally](https://github.com/alphagov/paas-product-pages?tab=readme-ov-file#local-development)
- check the [static build](https://github.com/alphagov/paas-product-pages?tab=readme-ov-file#review-static-build-production-build)

## Speed comparison (1 test)
Jest: [Time: 1.3 s](https://github.com/alphagov/paas-product-pages/actions/runs/8043216385/job/21964949932#step:5:25)
Vitest: [Duration 759ms](https://github.com/alphagov/paas-product-pages/actions/runs/8046101761/job/21972673446#step:5:23)
